x-pack/plugin/ccr/qa/src/main/java/org/elasticsearch/xpack/ccr/ESCCRRestTestCase.java

```diff
@@ -344,8 +345,14 @@ protected static void createAutoFollowPattern(RestClient client, String name, St
         assertOK(client.performRequest(request));
     }
 
+    private static final LazyInitializable<Long, RuntimeException> time = new LazyInitializable<>(System::currentTimeMillis);
+
     protected static String backingIndexName(String dataStreamName, int generation) {
-        return DataStream.getDefaultBackingIndexName(dataStreamName, generation);
+        return DataStream.getDefaultBackingIndexName(dataStreamName, generation, time.getOrCompute());
```

Comment:
Issue could be demonstrated by incrementing time by additional 24h on every call.